core/src/main/scala/matryoshka/AlgebraOps.scala

```diff
@@ -16,13 +16,27 @@
 
 package matryoshka
 
-import scalaz._
-import scalaz.syntax.comonad._
+import scalaz._, Scalaz._
 
-sealed class AlgebraOps[F[_], A](self: Algebra[F, A]) {
-  def generalize[W[_]: Comonad](implicit F: Functor[F]): GAlgebra[W, F, A] =
-    node => self(node ∘ (_.copoint))
+final class GElgotAlgebraMOps[E[_], G[_], M[_], F[_], A](
+  self: GElgotAlgebraM[E, G, M, F, A]) {
 
-  def generalizeElgot[W[_]: Comonad]: ElgotAlgebra[W, F, A] =
-    w => self(w.copoint)
+  def attribute(implicit E: Comonad[E], G: Comonad[G], M: Functor[M], F: Functor[F]) =
+    matryoshka.attribute[E, G, M, F, A](self)
+}
+ 
+final class GAlgebraMOps[G[_], M[_], F[_], A](self: GAlgebraM[G, M, F, A]) {
+  def generalizeElgot[E[_]: Comonad]: GElgotAlgebraM[E, G, M, F, A] =
```

Comment:
I tried this (fixing the final `E` to be `EE`), but ended up with a complaint that `f` is of type `E[F[G[A]]] => M[A]`, but expected `F[G[A]] => M[A]`.